src/NamingStrategy/PathNamingStrategy.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr\NamingStrategy;
+
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Will use the request path as filename.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class PathNamingStrategy implements NamingStrategyInterface
+{
+    /**
+     * @var array
+     */
+    private $options;
+
+    public function __construct(array $options = [])
+    {
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+        $this->options = $resolver->resolve($options);
+    }
+
+    public function name(RequestInterface $request): string
+    {
+        $parts = [$this->options['name_prefix']];
+
+        if ($this->options['hostname_prefix']) {
+            $parts[] = $request->getUri()->getHost();
+        }
+        $method = strtoupper($request->getMethod());
+
+        $parts[] = $method;
+        $parts[] = str_replace(\DIRECTORY_SEPARATOR, '_', trim($request->getUri()->getPath(), '/'));
```

Comment:
this needs to be `/`, as urls always have forward slash, even on windows. your code will break on windows (or create filenames with `/` in it if thats allowed, but that would be confusing and break when recordings are in git and later run on a different filesystem)